pysgrid/sgrid.py

```diff
@@ -514,14 +525,15 @@ def save_as_netcdf(self, filepath):
             # populate variables with data
             grid_time[:] = self.grid_times[:]
             grid_center_lon[:] = self.centers[..., 0]
-            grid_center_lat[:] = self.centers[..., 1] 
-            grid_node_lon[:] = self.nodes[..., 0]
-            grid_node_lat[:] = self.nodes[..., 1]
+            grid_center_lat[:] = self.centers[..., 1]
+            if self.nodes is not None: 
+                grid_node_lon[:] = self.nodes[..., 0]
+                grid_node_lat[:] = self.nodes[..., 1]
```

Comment:
>Consider using this syntax here and above (and possibly merging both codes into SGridND class making SGrid2D and SGrid3D simpler). It is virtually the same code, right? Just check for the dimensionality in SGridND and add a property there.

I'm not particularly inclined to spend too much on either `SGridND` or `SGrid3D` at this point. There are currently no compelling use-cases for the 3D grid. There have been suggestions to remove it from the SGrid standard, but a final decision hasn't been made yet. 

In the event that the 3D convention is retained, I would be more inclined to keep the `save_as_netcdf` methods within each class as they are with minor modification. The differences between the 2D and 3D grids are significant enough for that I don't moving things `SGridND` would reduce complexity, but rather just be moving it around. I would be open to taking the few common components and writing a method for those, in `SGridND`, but that would be the extent to which the code could be simplified.